report/4.Solution/BuiltInFunctions.tex

```diff
@@ -0,0 +1,58 @@
+\subsection{Built-in Functions}
+HCL features several built-in functions that map the languages functionality directly over to its c++ equivalent. 
+These functions are considered to be the most basic functionality of the language, and their use are often required in order to specify more complex functions. 
+This section describes the built-in functions of HCL, their implementation and their functionality
+
+\textbf{Build Functions}\\
+The implementation of all built-in functions are handled by the \texttt{buildFunction} function seen in snippet \ref{lis:buildFunction}.
+
+\begin{lstlisting}[language=Kotlin,label=lis:buildFunction,caption=The implementation of buildFunction.]
+private fun buildFunction(identifier: String, parameters: List<Parameter>, returnType: Type, body: String, attributes: LambdaExpressionAttributes = BuiltinLambdaAttributes) =
+    AstNode.Command.Declaration(returnType, identifier.asIdentifier(),
+	    AstNode.Command.Expression.LambdaExpression(
+            paramDeclarations = parameters.map {
+                AstNode.ParameterDeclaration(it.type, it.identifier.asIdentifier())
+            },
+            returnType = returnType,
+            attributes = attributes,
+            body = body.asRawCppLambdaBody()
+        )
+    )
+\end{lstlisting}
+The function returns a function-declaration with specified identifier, return-type and lambda-body.
+Where the lambda-body is composed of raw c++ code. 
```

Comment:
Think this line is dangling a bit, perhaps change the . of the previous line to a comma